src/angular/components/sso.component.ts

```diff
@@ -67,8 +67,23 @@ export class SsoComponent {
     }
 
     async submit(returnUri?: string, includeUserIdentifier?: boolean) {
-        const authorizeUrl = await this.buildAuthorizeUrl(returnUri, includeUserIdentifier);
-        this.platformUtilsService.launchUri(authorizeUrl, { sameWindow: true });
+        if (await this.preValidate()) {
```

Comment:
Generally, with anything network related like this API call, we associate this form with a promise and use `[appApiAction]="formPromise"` on the form itself so we can show a spinner on submission. You should be able to find multiple examples.

Reply:
I've updated the pattern